base/collect_database.go

```diff
@@ -115,6 +114,7 @@ func getElapsedTime(versionInfo map[string]interface{}, exp *Experiment) (int64,
 		case int64:
 			startingTime = rawStartingTime
 		case int:
+		case float64: // parsing the metrics file gives float64
```

Comment:
I think we want startingTime be of type string, and not any other type.
***
This way, it can be deserialized as follows ... 
```go

    // Declaring layout constant
    const layout = "Jan 2, 2006 at 3:04pm (MST)"
  
    // Calling Parse() method with its parameters
    startingTime, _ := time.Parse(layout, "Feb 4, 2014 at 6:05pm (PST)")
```

You can choose any layout of course, but it would probably best to begin with the default layout for time values in Golang. We can always adjust this part.
***

Wherever you compute elapsedTime, you can do:

```go
elapsedTime := startingTime.Sub(time.Now())
```

The above would make elapsedTime a [duration object](https://pkg.go.dev/time#Duration).

Reply:
> I think we want startingTime be of type string, and not any other type.

Do we have control over this? The type for VersionInfo is `[]map[string]interface{}`. I agree that it would be better if the type were well defined.